wpilibc/athena/src/DriverStation.cpp

```diff
@@ -584,9 +582,13 @@ void DriverStation::Run() {
   while (m_isRunning) {
     HAL_WaitForDSData();
     GetData();
+    // notify IsNewControlData variables
+    m_newControlData = true;
+
+    // notify WaitForData block
     {
       std::lock_guard<priority_mutex> lock(m_waitForDataMutex);
-      m_updatedControlLoopData = true;
+      m_waitForDataPredicate = true;
     }
     m_waitForDataCond.notify_all();
```

Comment:
Yeah. I've always found it weird how Java and C# will actually throw exceptions if you call notifyAll without holding the mutex.